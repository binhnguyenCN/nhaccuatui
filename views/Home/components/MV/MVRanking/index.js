import React from 'react';

import styles from './styles.module.scss';

const MVRanking = () => (
  <div className={styles.mvRanking}>
    <h1>MV RANKING</h1>
  </div>
);
export default MVRanking;
