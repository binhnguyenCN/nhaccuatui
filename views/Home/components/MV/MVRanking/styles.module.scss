.mv-ranking {
  height: 500px;
  background-color: green;
}
