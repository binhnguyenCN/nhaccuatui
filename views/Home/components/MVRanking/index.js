import React from 'react';

import styles from './styles.module.scss';

const MVRanking = () => (
  <div className={styles['mv-ranking']}>
    <h1>MV RANKING</h1>
  </div>
);
export default MVRanking;
