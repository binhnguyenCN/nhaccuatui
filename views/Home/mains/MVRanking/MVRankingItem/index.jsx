// libs
import Image from 'next/image';
import Link from 'next/link';
import React from 'react';

// layouts
import images from '../../../../../constants/images';
import styles from './styles.module.scss';

const MVRankingItem = ({ mv, index }) => (
  <li className={styles['mv-ranking-item']}>
    <div className={styles['mv-ranking-thumbnail']}>
      <Image
        src={mv.thumbnail}
        width={index === 1 ? 300 : 110}
        height={index === 1 ? 168 : 62}
        blurDataURL={images.blurDataURL}
        alt="mv ranking"
        title={mv.title}
      />
      <div className={styles['ranking-order']}>{index}</div>
      <div className={styles['thumbnail-dark-layer']} />
    </div>
    <div className={styles['mv-ranking-info']}>
      <div className={styles['mv-ranking-title']}>{mv.title}</div>
      <div className={styles['mv-ranking-artists']}>
        {mv.artists.map((artist, notFirstItem) => (
          <Link href={artist.imageUrl || '/'} key={artist.artistId}>
            {notFirstItem ? `, ${artist.name}` : artist.name}
          </Link>
        ))}
      </div>
    </div>
  </li>
);

export default MVRankingItem;
