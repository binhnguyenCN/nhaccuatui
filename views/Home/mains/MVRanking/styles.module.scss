.mv-ranking {
  grid-area: mvRanking;
  background-color: green;
}
