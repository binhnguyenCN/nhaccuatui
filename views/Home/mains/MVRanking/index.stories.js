import React from 'react';
import { videoList } from '../../../../api/mockApi';

import MVRanking from './index';

export default {
  title: 'Pages/Home/main/MVRanking',
  component: MVRanking,
  parameters: { mvs: videoList },
};
export const Template = () => <MVRanking mvs={videoList} />;
