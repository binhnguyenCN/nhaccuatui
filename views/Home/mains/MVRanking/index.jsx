// libs
import React from 'react';
// layouts
import CategoryTitle from '../../components/CategoryTitle';
import RankingList from '../../components/RankingList';
import RankingOption from '../../components/RankingOption';
import MVRankingItem from './MVRankingItem';
// others
import RankingOptions from '@/data_source/RankingOptions';
import CategoryTitles from '@/data_source/CategoryTitle';
import styles from './styles.module.scss';

const MVRanking = ({ mvs }) => (
  <div className={styles['mv-ranking-wrapper']}>
    <div className={styles['mv-ranking-inner']}>
      <CategoryTitle title={CategoryTitles.mvRanking} />
      <RankingOption options={RankingOptions.options} />
      <RankingList>
        {mvs &&
          mvs.map((mv, index) => {
            if (index < 5) {
              return (
                <MVRankingItem
                  title={mv.title}
                  thumbnail={mv.thumbnail}
                  artists={mv.artists}
                  index={index + 1}
                  key={mv.key}
                />
              );
            }
            return null;
          })}
      </RankingList>
    </div>
  </div>
);
export default MVRanking;
