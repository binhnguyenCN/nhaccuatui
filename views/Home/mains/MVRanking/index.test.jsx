// libs
import renderer from 'react-test-renderer';
// layouts
import MVRanking from '.';
// mocks
import mockData from '@/mocks/mockApi';

describe('render MV ranking', () => {
  it('should renders correctly', () => {
    // act
    const tree = renderer
      .create(<MVRanking mvs={mockData.videoList} />)
      .toJSON();
    // assert
    expect(tree).toMatchSnapshot();
  });
});
