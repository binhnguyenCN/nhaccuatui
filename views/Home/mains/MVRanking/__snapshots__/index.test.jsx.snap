// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render MV ranking should renders correctly 1`] = `
<div
  className="mv-ranking-wrapper"
>
  <div
    className="mv-ranking-inner"
  >
    <h3
      className="category-title"
    >
      <span
        href="/"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        <span>
          bxh mv
        </span>
        <span>
          <span
            aria-label="play-circle"
            className="anticon anticon-play-circle"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="play-circle"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm0 820c-205.4 0-372-166.6-372-372s166.6-372 372-372 372 166.6 372 372-166.6 372-372 372z"
              />
              <path
                d="M719.4 499.1l-296.1-215A15.9 15.9 0 00398 297v430c0 13.1 14.8 20.5 25.3 12.9l296.1-215a15.9 15.9 0 000-25.8zm-257.6 134V390.9L628.5 512 461.8 633.1z"
              />
            </svg>
          </span>
        </span>
      </span>
    </h3>
    <ul
      className="ranking-option"
    >
      <li
        className="ranking-item ranking-item-active"
      >
        <button
          className="ranking-btn"
          onClick={[Function]}
          type="button"
        >
          Việt Nam
        </button>
      </li>
      <li
        className="ranking-item"
      >
        <button
          className="ranking-btn"
          onClick={[Function]}
          type="button"
        >
          Âu Mỹ
        </button>
      </li>
      <li
        className="ranking-item"
      >
        <button
          className="ranking-btn"
          onClick={[Function]}
          type="button"
        >
          Hàn Quốc
        </button>
      </li>
    </ul>
    <ul
      className="ranking-list"
    >
      <li
        className="mv-ranking-item"
      >
        <div
          className="mv-ranking-thumbnail"
        >
          <img
            alt="mv ranking"
            height={168}
            src="https://avatar-ex-swe.nixcdn.com/mv/2022/04/22/d/8/1/d/1650617244595_640.jpg"
            title="Anh Lo Cho Em Hết"
            width={300}
          />
          <div
            className="ranking-order"
          >
            1
          </div>
          <div
            className="thumbnail-dark-layer"
          />
        </div>
        <div
          className="mv-ranking-info"
        >
          <div
            className="mv-ranking-title"
          >
            Anh Lo Cho Em Hết
          </div>
          <div
            className="mv-ranking-artists"
          >
            <a
              href="https://avatar-ex-swe.nixcdn.com/singer/avatar/2022/04/19/b/5/c/c/1650363937009_300.jpg"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Đàm Vĩnh Hưng
            </a>
            <a
              href="https://avatar-ex-swe.nixcdn.com/singer/avatar/2021/03/02/2/8/6/4/1614651512512_300.jpg"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              , Ricky Star
            </a>
          </div>
        </div>
      </li>
      <li
        className="mv-ranking-item"
      >
        <div
          className="mv-ranking-thumbnail"
        >
          <img
            alt="mv ranking"
            height={62}
            src="https://avatar-ex-swe.nixcdn.com/mv/2022/04/20/f/d/f/c/1650435244252_640.jpg"
            title="Quan Tâm Em Mỗi Ngày"
            width={110}
          />
          <div
            className="ranking-order"
          >
            2
          </div>
          <div
            className="thumbnail-dark-layer"
          />
        </div>
        <div
          className="mv-ranking-info"
        >
          <div
            className="mv-ranking-title"
          >
            Quan Tâm Em Mỗi Ngày
          </div>
          <div
            className="mv-ranking-artists"
          >
            <a
              href="https://avatar-ex-swe.nixcdn.com/singer/avatar/2019/07/22/f/e/a/2/1563760304939_300.jpg"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Hoàng Tôn
            </a>
          </div>
        </div>
      </li>
      <li
        className="mv-ranking-item"
      >
        <div
          className="mv-ranking-thumbnail"
        >
          <img
            alt="mv ranking"
            height={62}
            src="https://avatar-ex-swe.nixcdn.com/mv/2022/04/14/5/9/1/f/1649925760384_640.jpg"
            title="Tình Chết Trong Hững Hờ"
            width={110}
          />
          <div
            className="ranking-order"
          >
            3
          </div>
          <div
            className="thumbnail-dark-layer"
          />
        </div>
        <div
          className="mv-ranking-info"
        >
          <div
            className="mv-ranking-title"
          >
            Tình Chết Trong Hững Hờ
          </div>
          <div
            className="mv-ranking-artists"
          >
            <a
              href="https://avatar-ex-swe.nixcdn.com/singer/avatar/2019/09/17/7/2/a/2/1568707006619_300.jpg"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Lệ Quyên
            </a>
          </div>
        </div>
      </li>
      <li
        className="mv-ranking-item"
      >
        <div
          className="mv-ranking-thumbnail"
        >
          <img
            alt="mv ranking"
            height={62}
            src="https://avatar-ex-swe.nixcdn.com/mv/2022/04/19/3/c/c/7/1650344734007_640.jpg"
            title="Kết Thúc Như Bao Người"
            width={110}
          />
          <div
            className="ranking-order"
          >
            4
          </div>
          <div
            className="thumbnail-dark-layer"
          />
        </div>
        <div
          className="mv-ranking-info"
        >
          <div
            className="mv-ranking-title"
          >
            Kết Thúc Như Bao Người
          </div>
          <div
            className="mv-ranking-artists"
          >
            <a
              href="https://avatar-ex-swe.nixcdn.com/singer/avatar/2019/11/20/d/f/9/a/1574242685613_300.jpg"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Tăng Phúc
            </a>
          </div>
        </div>
      </li>
      <li
        className="mv-ranking-item"
      >
        <div
          className="mv-ranking-thumbnail"
        >
          <img
            alt="mv ranking"
            height={62}
            src="https://avatar-ex-swe.nixcdn.com/mv/2022/03/24/4/b/3/6/1648109917566_640.jpg"
            title="Chống Lại Thế Giới"
            width={110}
          />
          <div
            className="ranking-order"
          >
            5
          </div>
          <div
            className="thumbnail-dark-layer"
          />
        </div>
        <div
          className="mv-ranking-info"
        >
          <div
            className="mv-ranking-title"
          >
            Chống Lại Thế Giới
          </div>
          <div
            className="mv-ranking-artists"
          >
            <a
              href="https://avatar-ex-swe.nixcdn.com/singer/avatar/2020/12/15/e/f/5/9/1608025280378_300.jpg"
              onClick={[Function]}
              onMouseEnter={[Function]}
            >
              Ngô Kiến Huy
            </a>
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
